sdk/go/common/workspace/project.go

```diff
@@ -358,16 +358,16 @@ func ValidateProject(raw interface{}) error {
 	var errs *multierror.Error
 	var appendError func(err *jsonschema.ValidationError)
 	appendError = func(err *jsonschema.ValidationError) {
-		if err.InstanceLocation != "" && err.Message != "" {
+		if err.InstanceLocation != "" && err.Message != "" && len(err.Causes) == 0 {
```

Comment:
We can already limit noise by not printing an error's message when it contains nested errors, as the nested errors typically are the root cause, i.e.
```
error: could not get cloud url: could not load current project: could not validate '/Users/alexqiu/github.com/pulumi/test-296/Pulumi.yaml': 8 errors occurred:
        * #/config/bucketNamePrefix: oneOf failed
        * #/config/bucketNamePrefix: expected string, but got object
        * #/config/bucketNamePrefix: expected integer, but got object
        * #/config/bucketNamePrefix: expected boolean, but got object
        * #/config/bucketNamePrefix: expected array, but got object
        * #/config/bucketNamePrefix: doesn't validate with '/$defs/configTypeDeclaration'
        * #/config/bucketNamePrefix/type: doesn't validate with '/$defs/simpleConfigType'
        * #/config/bucketNamePrefix/type: value must be one of "string", "integer", "boolean", "array"
```
becomes
`5 errors occurred:
        * #/config/bucketNamePrefix: expected string, but got object
        * #/config/bucketNamePrefix: expected integer, but got object
        * #/config/bucketNamePrefix: expected boolean, but got object
        * #/config/bucketNamePrefix: expected array, but got object
        * #/config/bucketNamePrefix/type: value must be one of "string", "integer", "boolean", "array"`